crates/re_build_search_index/src/ingest/rust.rs

```diff
@@ -0,0 +1,512 @@
+        &mut self,
+        pub_in_priv: bool,
+        type_id: &ItemId,
+        id: &ItemId,
+        parent_kind: ParentItemKind,
+    ) {
+        let item = &self.krate.index[id];
+        let kind = match &item.inner {
+            ItemEnum::Function(_) => ItemKind::Assoc(parent_kind, AssocItemKind::Method),
+            ItemEnum::AssocConst { .. } => {
+                ItemKind::Assoc(parent_kind, AssocItemKind::AssociatedConstant)
+            }
+            ItemEnum::AssocType { .. } => {
+                ItemKind::Assoc(parent_kind, AssocItemKind::AssociatedType)
+            }
+            _ => unreachable!("invalid associated item {item:#?}"),
+        };
+
+        let name = item.name.as_ref().unwrap().clone();
+        let path = self.resolve_path(pub_in_priv, type_id).with_item(name);
+        self.push_with_path(id, kind, &path);
+    }
+
+    fn resolve_path(&self, pub_in_priv: bool, id: &ItemId) -> Vec<String> {
+        if pub_in_priv {
+            let name = self.krate.index[id].name.as_ref().unwrap().clone();
+            self.module_path.with_item(name)
+        } else {
+            let Some(summary) = self.krate.paths.get(id) else {
+                panic!(
+                    "expected item {id:?} to have a rustdoc-generated path (module_path={:?})",
+                    self.module_path
+                );
+            };
+            summary.path.clone()
+        }
+    }
+}
+
+fn base_url(_version: &Version, krate: &Crate) -> String {
+    // format!("https://docs.rs/{krate_name}/{version}")
+    format!("https://docs.rs/{}/latest", krate.name())
+}
+
+fn document(path: String, url: String, docs: String) -> DocumentData {
+    DocumentData {
+        kind: DocumentKind::Rust,
+        title: path,
+        hidden_tags: vec!["rust".into()],
+        tags: vec![],
+        content: docs,
+        url,
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+enum ItemKind {
+    Module,
+    Struct,
+    Enum,
+    Trait,
+    Function,
+    Type,
+    Constant,
+    Macro,
+    Assoc(ParentItemKind, AssocItemKind),
+}
+
+impl Display for ItemKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s = match self {
+            ItemKind::Module => "module",
+            ItemKind::Struct => "struct",
+            ItemKind::Enum => "enum",
+            ItemKind::Trait => "trait",
+            ItemKind::Function => "fn",
+            ItemKind::Type => "type",
+            ItemKind::Constant => "constant",
+            ItemKind::Macro => "macro",
+            ItemKind::Assoc(ParentItemKind::Trait, AssocItemKind::Method) => "tymethod",
+            ItemKind::Assoc(_, AssocItemKind::Method) => "method",
+            ItemKind::Assoc(_, AssocItemKind::AssociatedConstant) => "associatedconstant",
+            ItemKind::Assoc(_, AssocItemKind::AssociatedType) => "associatedtype",
+        };
+        f.write_str(s)
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+enum AssocItemKind {
+    Method,
```

Comment:
Is this pertaining to associated methods, as documented in https://doc.rust-lang.org/rust-by-example/fn/methods.html ?

Reply:
Yes. Associated items have different paths in docs.rs, so they get a separate kind enum